extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/export/PrometheusMeterRegistryProducer.java

```diff
@@ -1,25 +1,33 @@
 package io.quarkus.micrometer.runtime.export;
 
+import javax.enterprise.inject.Alternative;
+import javax.enterprise.inject.Instance;
 import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
 import javax.inject.Singleton;
 import javax.interceptor.Interceptor;
 
 import io.micrometer.core.instrument.Clock;
 import io.micrometer.prometheus.PrometheusConfig;
 import io.micrometer.prometheus.PrometheusMeterRegistry;
 import io.prometheus.client.CollectorRegistry;
-import io.quarkus.arc.AlternativePriority;
+import io.prometheus.client.exemplars.ExemplarSampler;
+import io.quarkus.arc.Priority;
 
 /**
  * This producer is only registered if the {@code quarkus.micrometer.export.prometheus.default-registry} is set to {@code true}.
  */
 public class PrometheusMeterRegistryProducer {
 
+    @Inject
+    Instance<ExemplarSampler> sampler;
+
     @Produces
     @Singleton
-    @AlternativePriority(Interceptor.Priority.APPLICATION + 100)
+    @Alternative
+    @Priority(Interceptor.Priority.APPLICATION + 100)
     public PrometheusMeterRegistry registry(PrometheusConfig config, CollectorRegistry collectorRegistry, Clock clock) {
-        return new PrometheusMeterRegistry(config, collectorRegistry, clock);
+        return new PrometheusMeterRegistry(config, collectorRegistry, clock, sampler.stream().findFirst().orElse(null));
```

Comment:
This is also just getting the trace id stored in the current context... This will not contain the trace id for the request that had the min or max.

Reply:
that isn't how this works. You always capture the span active for this measurement. The rest is sorted out later.

Reply:
our responsibility is only to capture the id for the current request. As the code being instrumented, we have no ability to know what the min/max observed is (and if we did, we would be duplicating what the registry/backend is already doing.. )

Reply:
So am I write when I understand, I keep this approach (except for the Optional thing) ?

Reply:
yes. I think so. I'm chewing on the Vertx HTTP thing.. that timing is annoying..